lib/tempfile.c

```diff
@@ -42,8 +43,15 @@ FILE *tempfile(void)
 	oldmask = umask(0077);
 	fd = open(_PATH_TMP, O_TMPFILE | O_RDWR | O_EXCL | O_CLOEXEC, S_IRUSR | S_IWUSR);
 	umask(oldmask);
-	if (-1 == fd)
-		return NULL;
+	if (fd == -1) {
+            if (errno == EOPNOTSUPP) {
+                // Fall back to tmpfile() if O_TMPFILE is not supported
+                return tmpfile();
+            } else {
+                // Handle other errors, return NULL
+                return NULL;
+            }
```

Comment:
Comments on added block:

 - Else not needed when if has return.  
 - Please use C style comments `/* .. */`

I.e., rewrite as:

```c
  /* Fall back to tmpfile() if O_TMPFILE is not supported */
  if (errno == EOPNOTSUPP)
      return tmpfile();
  
  return NULL;  
```

Reply:
change done, thanks for feedback.